src/utils.ts

```diff
@@ -203,6 +203,6 @@ export function impProto(options: Options, module: string, type: string): Import
   if (options.onlyTypes) {
     return imp('t:' + importString);
   } else {
-    return imp(importString);
+    return imp(importString + '.js');
```

Comment:
Huh! I didn't know that if you have a file called `foo.ts`, you're allowed to import it as `foo.js`.

That makes sense, I guess, since it will all wash out to `*.js` files in the end / post-compilation.

I dunno, that said, does it work to make this `.ts`? 

Personally I think that'd be less surprising to the reader, since they would normally be reading the files as pre-`tsc` TS code. And I assume `tsc` will make all of the `foo.ts` imports into `foo.js` in the output anyway (?).

Reply:
> does it work to make this .ts?

No, this returns an error from the compiler (counter-intuitively)

Reply:
Also huh!

Wdyt about doing a `s,\./_/` on the file names, so that your `myother.pb.ts` files end up as `myother_bs` and imports like like:

```
import { MyMessage } from '../myother/myother_pb'
```

It seems like this would be less disruptive to existing users that aren't use the `.pb` naming convention.

Granted, it sounds like file extensions are becoming defacto required in ESM, but hopefully/surely they've fixed/allow `.ts` extensions by then.

Reply:
I'm very interested in using .pb.js since I'm using .pb.go - and with this patch it's working correctly.

We're going to have to use the .js suffix for all non-package imports with ESM anyway - (for example protobufjs/minimal.js) so eventually this will be required. And already the TypeScript compiler ignores the .js suffix and finds the .ts file, so it shouldn't be backwards incompatible, right?

If it causes an issue I guess it could be another option (to add .js as the suffix everywhere we try to import a ts file).

Reply:
Yeah, totally get/agree we'll need it with ESM, albeit I'm ~80% sure that `*.ts` suffixes will be/are kosher with that tsconfig (I've only seen the TS release notes go by, haven't personally used it yet).

Maybe do a conditional here, like if `importString` has a `.` in it, add the `.js` export?

I'm just thinking, while this makes a ton of sense for you, for the other pre-ESM users of ts-proto, they might read the imports and just do an raised eyebrow about "why is that there?", even if I agree it won't break anything for them.

If the conditional-if-`.`-in-`importString` fixes your `.pb.js` use case, let's do that? But if not, yeah, I'm good with always having it.

Reply:
I agree it's confusing and at first when I saw the .js imports in files I was like.. wat?

But like you said, it's the way the TypeScript developers decided to go for this, and I guess it's slightly less ambiguous than guessing to add the .ts suffix.

Reply:
With regards to importing with '.ts' suffix: https://github.com/microsoft/TypeScript/issues/37582

Reply:
Nice! Thanks for the link to the TS issue re `.ts` imports.